src/xenia/gpu/vulkan/texture_cache.cc

```diff
@@ -1078,10 +1078,17 @@ bool TextureCache::UploadTexture(VkCommandBuffer command_buffer,
   if (dest->format == VK_FORMAT_D16_UNORM_S8_UINT ||
       dest->format == VK_FORMAT_D24_UNORM_S8_UINT ||
       dest->format == VK_FORMAT_D32_SFLOAT_S8_UINT) {
-    // Do just a depth upload (for now).
-    // This assumes depth buffers don't have mips (hopefully they don't)
-    assert_true(src.mip_levels() == 1);
-    copy_regions[0].imageSubresource.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+    // Depth buffers can have mips, so dont's assert this:
+    // assert_true(src.mip_levels() == 1);
+    if (src.mip_levels() == 1) {
```

Comment:
The if is not required, just the for loop gives the exact same result

Reply:
I know, but i kept the original code in there for reference and in case you want to do something special in the case there is only a single level in there. I can imagine that in some cases you want to do something different, so this allows that.

Reply:
"Keeping the code for reference" is what git does for you, no need to keep code around which doesn't do anything.
If you want to clarify the for loop, add a comment stating that i = 0 is different from the rest, and/or why.
If someone wants to implement something different for i=0, he can change the code again later

Reply:
Ok clear. I'll just remove it for performance reasons so that another developer can always try to do something different :). So will commit your requests here.

Reply:
This line will fail:

 if (src.mip_levels() == 1) 

src.mip_levels() must be stored in a variable first.

Reply:
Ok fixed with my latest commit.